rest_framework/authentication.py

```diff
@@ -85,7 +85,16 @@ def authenticate_credentials(self, userid, password):
         """
         Authenticate the userid and password against username and password.
         """
-        user = authenticate(username=userid, password=password)
+        user_model = get_user_model()
+        if hasattr(user_model, 'USERNAME_FIELD'):
+            username_field = user_model.USERNAME_FIELD
+        else:
+            username_field = 'username'
```

Comment:
We could slim this down with `username_field = getattr(user_model, 'USERNAME_FIELD', 'username')` I think?